bftengine/src/bftengine/messages/PrePrepareMsg.cpp

```diff
@@ -117,7 +117,6 @@ void PrePrepareMsg::addRequest(const char* pRequest, uint32_t requestSize) {
 void PrePrepareMsg::finishAddingRequests() {
   Assert(!isNull());
   Assert(!isReady());
-  Assert(b()->numberOfRequests > 0);
```

Comment:
I don't think you should remove this Assert. This method should not be called with no pre-prepares. Even in the code you modified you return if there are no requests before finalizing. Otherwise, we enter this code, and then immediately send a `PrePrepareMsg` with no actual requests. In that case we don't want to send the PrePrepareMsg at all. There are of course other ways to perform this logic, but it seems that the rest of the code assumes this function is a gate to prevent generating useless PrePrepareMsgs and then sending them.

Reply:
I understand. thanks